sql/src/main/java/org/apache/apex/malhar/sql/codegen/BeanClassGenerator.java

```diff
@@ -0,0 +1,796 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.sql.codegen;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.codehaus.jettison.json.JSONException;
+
+import org.apache.apex.malhar.sql.schema.TupleSchemaRegistry;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.xbean.asm5.ClassWriter;
+import org.apache.xbean.asm5.Opcodes;
+import org.apache.xbean.asm5.tree.ClassNode;
+import org.apache.xbean.asm5.tree.FieldInsnNode;
+import org.apache.xbean.asm5.tree.FieldNode;
+import org.apache.xbean.asm5.tree.InsnNode;
+import org.apache.xbean.asm5.tree.IntInsnNode;
+import org.apache.xbean.asm5.tree.JumpInsnNode;
+import org.apache.xbean.asm5.tree.LabelNode;
+import org.apache.xbean.asm5.tree.LdcInsnNode;
+import org.apache.xbean.asm5.tree.MethodInsnNode;
+import org.apache.xbean.asm5.tree.MethodNode;
+import org.apache.xbean.asm5.tree.TypeInsnNode;
+import org.apache.xbean.asm5.tree.VarInsnNode;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+/**
+ * Creates a bean class on fly.
+ */
+@InterfaceStability.Evolving
+public class BeanClassGenerator
```

Comment:
Sorry for the long text. but after spending small time I have following information about cglib::

Our requirement:
1. Should be able to create a bean/POJO.
2. Write to jar
3. After jar is present in classpath, should be able to create an object out of it using new or newInstance().

What cglib provides:
The API for cglib are indeed very easy to use.
Looking internally into cglib, it does use ASM5 library to generate the class but also adds some extra callback handlers to it.
From this particular post (http://stackoverflow.com/questions/35146746/write-java-object-into-class-file), we can create a byte[] of generated class to be written to a jar for loading in yarn containers, but along with that it also needs cglib to work. i.e. The generated class is not equivalent of POJO having only private fields+public getters and setters. The bean is not standalone but it contains something more.

Moreover, the bean generated by cglib should be created using BeanGenerator.create method which directly gives a object of the class.

I haven't found any example over the web which uses cglib to generate the bean, write to jar, load jar in different program and use the bean. 
As long all of that is part of the same jvm, cglib seems to be a really good option. But because our functionality is across JVM (apex cli will generate POJOs and those will be used by containers in yarn cluster), my evaluation about this library is not as good option as what we have (BeanClassGenerator).

Please correct me if I'm wrong anywhere.